### PR TITLE
Update/patternception saving bug tests

### PR DESCRIPTION
**Pull request by johnstonphilip**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/studiopress/pattern-manager/wiki/For-Developers#contributor-license-agreement-cla) with WP Engine.

### Summary of changes
This is a PR to #177 which simply adds some PHPunit tests for the changes in question there.

### CODE REVIEW

**Comment by johnstonphilip** [comment]
I'm not sure why the diff is showing changes outside phpunit changes, but that's all this one is.

**Comment by kienstra** [comment]
Thanks, I'll review this today.

File: wp-modules/editor/tests/ModelTest.php [hunk 1]
```diff
@@ -81,4 +81,103 @@ public function test_delete_pattern_posts_correct_post() {
...
+
+		// Mock a post object so we can test modifying only the content.
+		$post_id            = -997; // negative ID, to avoid clash with a valid post.
+		$post               = new \stdClass();
+		$post->ID           = $post_id;
+		$post->post_author  = 1;
+		$post->post_title   = $pattern['title'];
+		$post->post_content = 'This is modified pattern content, but nothing else is changed!';
+		$post->post_status  = 'publish';
+		$post->post_name    = $pattern['slug'];
```
Comment (kienstra) [inline wp-modules/editor/tests/ModelTest.php:161]: Nitpick:
```suggestion
		$post->post_name    = 'mismatched-name';
```

**Review by kienstra — APPROVED** [review]
Hi @johnstonphilip,
Really nice work.

Some comments below, but no blocker to merging.

File: wp-modules/editor/tests/ModelTest.php [hunk 2]
```diff
@@ -81,4 +81,103 @@ public function test_delete_pattern_posts_correct_post() {
...
+		// Convert to WP_Post object.
+		$wp_post = new \WP_Post( $post );
+
+		// Save the mocked pattern to the disk.
+		\PatternManager\Editor\save_pattern_to_file( $wp_post );
+
+		// Get the contents of the file that was saved.
+		$pattern = \PatternManager\PatternDataHandlers\get_pattern_by_name( $post->post_name );
+
+		// Make sure the slug of the post and the slug in the file match.
```
[the remark this answers is not in the document]
Comment (kienstra) [reply wp-modules/editor/tests/ModelTest.php:111]: Well, the `->post_name` and slug are the same, so maybe this is fine.

File: wp-modules/editor/tests/ModelTest.php [hunk 3]
```diff
@@ -81,4 +81,103 @@ public function test_delete_pattern_posts_correct_post() {
 			get_posts( [ 'post_type' => 'pm_pattern ' ] )
 		);
 	}
+
+	/**
+	 * Tests that a pattern newly created with Pattern Manager results in a slug that matches the title.
+	 */
+	public function test_new_pattern_title_matches_slug() {
```
[the remark this answers is not in the document]
Comment (johnstonphilip) [reply wp-modules/editor/tests/ModelTest.php:88]: Yeah if you don't mind, I'd love to fully understand how this works :)

[the remark this answers is not in the document]
Comment (kienstra) [reply wp-modules/editor/tests/ModelTest.php:88]: Sure, I'll do that. Still, don't wait to merge this.

File: wp-modules/editor/tests/ModelTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnstonphilip) [reply wp-modules/editor/tests/ModelTest.php:161]: Is there a reason you'd set the slug to this as opposed to pulling from the file?

File: wp-modules/editor/tests/ModelTest.php [hunk 4]
```diff
@@ -81,4 +81,103 @@ public function test_delete_pattern_posts_correct_post() {
...
+		$post->post_author  = 1;
+		$post->post_title   = 'New Pattern, originally created with Pattern Manager.';
+		$post->post_content = 'test pattern content';
+		$post->post_status  = 'publish';
+		$post->post_name    = 'new-pattern-originally-created-with-pattern-manager';
+		$post->post_type    = get_pattern_post_type();
+		$post->filter       = 'raw';
+
+		// Convert to WP_Post object.
+		$wp_post = new \WP_Post( $post );
```
[the remark this answers is not in the document]
Comment (johnstonphilip) [reply wp-modules/editor/tests/ModelTest.php:103]: Oh nice!

File: wp-modules/editor/tests/ModelTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnstonphilip) [reply wp-modules/editor/tests/ModelTest.php:111]: Thanks WordPress! :D

File: wp-modules/editor/tests/ModelTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (kienstra) [reply wp-modules/editor/tests/ModelTest.php:161]: It's a nitpick 😄 

But if it were to come from the file, it should be the `$pattern['name']`.

In tests, I like setting values to string literals when possible, as strings in variables are mutable.

But I'm really nitpicking here 😄

**Pull request merged**